xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java

```diff
@@ -374,8 +395,18 @@ public void update(DocumentReference documentReference, Map<String, String> para
             // More information: https://jira.xwiki.org/browse/XWIKI-13493
             if (!parentChildConfiguration.isParentChildMechanismEnabled()) {
                 DocumentReference hierarchicalParent = getHierarchicalParent(documentReference);
-                if (!hierarchicalParent.equals(document.getParentReference())) {
-                    document.setParentReference(hierarchicalParent);
+                DocumentReference oldParent = document.getParentReference();
+
+                DocumentReference hierarchicalParentDoc = this.entityReferenceDocumentReferenceResolver
+                    .resolve(hierarchicalParent);
+                DocumentReference oldParentDoc = this.entityReferenceDocumentReferenceResolver.resolve(oldParent);
```

Comment:
A ``DocumentReference`` instance is **always complete**. It can't have missing parts so resolving it it doesn't really make sense. Check the JavaDoc of [``DefaultReferenceDocumentReferenceResolver``](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-model/src/main/java/org/xwiki/model/internal/reference/DefaultReferenceDocumentReferenceResolver.java#L33)

> from Entity Reference (when they miss some parent references or have NULL values)